src/main/java/edu/harvard/iq/dataverse/api/imports/ImportServiceBean.java

```diff
@@ -268,116 +275,132 @@ public Dataset doImportHarvestedDataset(DataverseRequest dataverseRequest, Harve
+                // Creating a new dataset from scratch:
+                
+                harvestedDataset = parser.parseDataset(obj);
+                harvestedDataset.setOwner(owner);
+                
+                harvestedDataset.setHarvestedFrom(harvestingClient);
+                harvestedDataset.setHarvestIdentifier(harvestIdentifier);
+                
+                harvestedVersion = harvestedDataset.getVersions().get(0);
+            } else {
+                // We already have a dataset with this id in the database.
+                
+                // If this dataset already exists IN ANOTHER COLLECTION
+                // we are just going to skip it!
+                if (existingDataset.getOwner() != null && !owner.getId().equals(existingDataset.getOwner().getId())) {
+                    throw new ImportException("The dataset with the global id " + globalIdString + " already exists, in the dataverse " + existingDataset.getOwner().getAlias() + ", skipping.");
+                }
+                // And if we already have a dataset with this same id, in this same
+                // dataverse, but it is a LOCAL dataset (can happen!), we're going to 
+                // skip it also: 
+                if (!existingDataset.isHarvested()) {
+                    throw new ImportException("A LOCAL dataset with the global id " + globalIdString + " already exists in this dataverse; skipping.");
+                }
+                // For harvested datasets, there should always only be one version.
+                // We will replace the current version with the imported version.
+                // @todo or should we just destroy any extra versions quietly?
+                if (existingDataset.getVersions().size() != 1) {
+                    throw new ImportException("Error importing Harvested Dataset, existing dataset has " + existingDataset.getVersions().size() + " versions");
+                }
+                                
+                harvestedVersion = parser.parseDatasetVersion(obj.getJsonObject("datasetVersion"));
+                Dataset tempDataset = createTemporaryHarvestedDataset(harvestedVersion);
+                // Temporarily attach the temporary dataset to the parent Collection: 
+                // (this will be needed for the purposes of looking up the configured
+                // metadatablocks and such)
+                tempDataset.setOwner(owner);
+            }
+        
+            // Either a full new import, or an update of an existing harvested
+            // Dataset, perform some cleanup on the new version imported from the 
+            // parsed metadata:
+            
+            harvestedVersion.setDatasetFields(harvestedVersion.initDatasetFields());
 
-            if (ds.getVersions().get(0).getReleaseTime() == null) {
-                ds.getVersions().get(0).setReleaseTime(oaiDateStamp);
+            if (harvestedVersion.getReleaseTime() == null) {
```

Comment:
So don't update a current existing version's releaseTime if the oaiDateStamp is different? (I'm assuming a new one just parsed is always null? Or could there be a newly parsed version where releaseTime and oaiDateStamp are different?)

Reply:
Let me check, but I don't think it's always null - if it's a harvest from another Dataverse using our proprietary json format, I would expect it to be the actual releaseTime from the source version.

Reply:
(that said, this was one of the few lines I just adopted intact from the old code without thinking about it much)

Reply:
Interesting. The json export as supplied by the source Dataverse does contain the `"releaseTime"` timestamp. However, our  `parseDatasetVersion()` method in `JsonParser` does this: 
 ```dsv.setReleaseTime(parseDate(obj.getString("releaseDate", null)));```

I can't think of where this `releaseDate` token comes from right away. The export method in `JsonPrinter` does 

```add("releaseTime", format(dsv.getReleaseTime()))```

and it never outputs `releaseDate`. 🤔

Reply:
Haven't tested, but we may be getting a non-null releaseTime from the remote Dataverse when harvesting using DDI as the format. (It looks like the json produced from DTO produced from our OAI_DDI will have this `"releaseDate"` field. 🤯

Reply:
All that aside, the `oaiDateStamp` is a sensible substitute for `releaseTime`. Since that is what the remote archive advertises as the date the record in question was last modified and/or made available - or _released_ via OAI.